src/components/Tables/Events.tsx

```diff
@@ -1,84 +1,170 @@
+    const { mutate: deleteMutation, isSuccess: deleteMutationIsSuccess } =
+        api.event.delete.useMutation()
 
-    const columns = useMemo<ColumnDef<EventWithArtistVenue>[]>(
+    const handleEditClick = useCallback(
+        async (event: EventWithArtistVenue) => {
+            const params = new URLSearchParams()
+            params.set('id', event.id)
+            await router.push(
+                {
+                    pathname: '/admin/edit-event',
+                    query: params.toString()
+                },
+                undefined,
+                {
+                    shallow: true
+                }
+            )
+        },
+        [router]
+    )
+
+    const handleToggleFeatured = useCallback(
+        (event: EventWithArtistVenue) => {
+            setFeaturedMutation(
+                { id: event.id },
+                {
+                    onSuccess: () => {
+                        void refetch()
+                    },
+                    onError: (e) => {
+                        setError(
+                            'Setting featured event failed. Please try again later.'
+                        )
+                        console.error(e)
+                    }
+                }
+            )
+        },
+        [setFeaturedMutation, refetch]
+    )
+
+    const handleDelete = useCallback(
+        (event: EventWithArtistVenue) => {
+            deleteMutation(
+                { id: event.id },
+                {
+                    onSuccess: () => {
+                        void refetch()
+                    },
+                    onError: (e) => {
+                        setError('Delete venue failed. Please try again later.')
+                        console.error(e)
+                    }
+                }
+            )
+        },
+        [deleteMutation, refetch]
+    )
+
+    const columns = useMemo(
         () => [
-            {
-                accessorKey: 'name',
+            columnHelper.accessor((row) => row.name, {
                 cell: (info) => info.getValue(),
-                header: () => <span>Event</span>
-            },
-            {
-                accessorKey: 'venue.name',
+                header: 'Event'
+            }),
+            columnHelper.accessor((row) => row.venue.name, {
                 cell: (info) => info.getValue(),
-                header: () => <span>Venue</span>
-            },
-            {
-                accessorKey: 'startDate',
-                cell: (info) =>
-                    format(new Date(info.getValue() as string), 'h:mm a'),
-                header: () => <span>Start</span>,
+                header: 'Venue'
+            }),
+            columnHelper.accessor((row) => row.startDate, {
+                cell: (info) => {
+                    const content = info.getValue()
+                    if (content) {
+                        format(
+                            new Date(info.getValue() as unknown as string),
```

Comment:
Please adjust this and the column below in accordance with the feedback on the other column with this aliasing